src/framework/mlt_producer.c

```diff
@@ -1291,8 +1291,11 @@ void mlt_producer_set_creation_time( mlt_producer self, int64_t creation_time )
 
 int mlt_producer_probe( mlt_producer self )
 {
-	if ( self && self->probe )
+	if ( self )
 	{
-		return self->probe( self );
+		int ( *probe )( mlt_producer ) = mlt_properties_get_data( MLT_PRODUCER_PROPERTIES(self), "_probe", NULL );
```

Comment:
To make this property name slightly cleaner while still mitigate name conflict:
a) The leading underscore is not needed or that useful on data properties that have no serializer. These are naturally already hidden.
b) Prefix the property name like we do for programming: `mlt_producer_probe` 
What do you think?

Reply:
That is a good idea. Done.